### PR TITLE
Remove coverage profile from service pom

### PR DESCRIPTION
**Pull request by srnagar**
This PR updates the service poms to remove the `coverage` profile. The pom file required to create aggregate coverage report per service is [now generated](https://github.com/Azure/azure-sdk-for-java/pull/29035) in the pipeline. So, the service poms can go back to being track agnostic and contain only the list of modules.

### CODE REVIEW

**Review by alzimmermsft — APPROVED** [review]

**Review by digimaun — APPROVED** [review]

**Review by weidongxu-microsoft — APPROVED** [review]

**Review by saragluna — CHANGES_REQUESTED** [review]

File: sdk/spring-experimental/pom.xml [hunk 1]
```diff
@@ -8,57 +8,8 @@
...
-    </profile>
-
-    <profile>
-      <id>dev</id>
-      <modules>
-        <module>../../eng/code-quality-reports</module>
-        <module>spring-cloud-azure-native-configuration</module>
-      </modules>
-    </profile>
-  </profiles>
```
[the remark this answers is not in the document]
Comment (srnagar) [reply sdk/spring-experimental/pom.xml]: Yes, I have add the dev and default profiles back.

**Comment by JimSuplizio** [comment]
I'm going to override check enforcer. The resourcemanagerhybrid libraries and incorrect triggers caused the pipeline that should not have run to get run. Those triggers were fixed as part of this PR (thank you @srnagar)

File: sdk/spring/scripts/compatibility_insert_dependencymanagement.py [hunk 2]
```diff
@@ -101,6 +101,10 @@ def add_dependency_management_for_file(file_path, spring_boot_dependencies_versi
     with open(file_path, 'r', encoding = 'utf-8') as pom_file:
         pom_file_content = pom_file.read()
         insert_position = pom_file_content.find('<dependencies>')
+        if(insert_position == -1):
+            # no dependencies section in pom, not adding <dependencyManagement> section
+            print("No dependencies section found in " + file_path + ". Not adding dependencyManagement.")
+            return
```
Comment (saragluna) [inline sdk/spring/scripts/compatibility_insert_dependencymanagement.py:107]: Thanks for this change!

**Review by saragluna — APPROVED** [review]
LGTM for Spring changes.

**Pull request merged**